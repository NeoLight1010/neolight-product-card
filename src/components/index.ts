import { ProductCard as ProductCardHOC } from './ProductCard';

import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';
import { ProductButtons } from './ProductButtons';

export { ProductImage } from './ProductImage';
export { ProductTitle } from './ProductTitle';
export { ProductButtons } from './ProductButtons';

type HOCType = typeof ProductCardHOC;
interface ProductCardHOCInterface extends HOCType {
    Image: typeof ProductImage,
    Title: typeof ProductTitle,
    Buttons: typeof ProductButtons,
}

export const ProductCard: ProductCardHOCInterface = Object.assign(
  ProductCardHOC,
  {
    Image: ProductImage,
    Title: ProductTitle,
    Buttons: ProductButtons,
  }
);

export default ProductCard;
