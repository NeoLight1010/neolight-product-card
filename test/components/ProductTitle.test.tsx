import React from "react"; React;

import renderer from "react-test-renderer";
import { ProductTitle, ProductCard } from "../../src/components";
import {product1} from "../data/products";

describe("Product Title", () => {
    it("should show component with custom title", () => {
        const wrapper = renderer.create(
            <ProductTitle title="Custom Product" />
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it("should show component with the product's name", () => {
        const wrapper = renderer.create(
            <ProductCard product={product1}>
                {() => (
                    <ProductTitle />
                )}
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
