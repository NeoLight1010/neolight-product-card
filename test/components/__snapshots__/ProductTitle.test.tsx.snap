// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Title should show component with custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;

exports[`Product Title should show component with the product's name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffe Mug (no image)
  </span>
</div>
`;
